net/api/client/client_test.go

```diff
@@ -210,8 +210,7 @@ func TestClient_AddRecord(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	creds := thread.NewDefaultCreds(id, thread.WithPrivKey(authorSk))
-	_, err = client.CreateThread(context.Background(), creds, core.WithThreadKey(tk), core.WithLogKey(logPk))
+	_, err = client.CreateThread(context.Background(), id, core.WithThreadKey(tk), core.WithLogKey(logPk), core.WithNewThreadIdentity(authorSk))
```

Comment:
Here's how a client would use their own identity provider with a request. If they aren't comfortable trusting the client code to not suck up their private key, they could implements a wrapper that also implements `Identity`, and keep private key private. In any case, private key is sent anywhere.